symfony/plugins/orangehrmTimePlugin/service/CustomerService.php

```diff
@@ -17,68 +17,97 @@
  * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
  * Boston, MA  02110-1301, USA
  */
-class CustomerService extends BaseService {
 
-    private $customerDao;
+namespace OrangeHRM\Time\Service;
+
+use Exception;
+use OrangeHRM\Entity\Customer;
+use OrangeHRM\Time\Dao\CustomerDao;
+use OrangeHRM\Time\Dto\CustomerSearchFilterParams;
+use OrangeHRM\Core\Exception\ServiceException;
+
+
+class CustomerService
+{
+
+    /**
+     * @var CustomerDao|null
+     */
+    private ?CustomerDao $customerDao = null;
 
     /**
      * Construct
      */
-    public function __construct() {
+    public function __construct()
+    {
         $this->customerDao = new CustomerDao();
     }
 
     /**
      *
-     * @return <type>
+     * @return CustomerDao
      */
-    public function getCustomerDao() {
+    public function getCustomerDao(): CustomerDao
+    {
+        if (!$this->customerDao instanceof CustomerDao) {
+            $this->customerDao = new CustomerDao();
+        }
         return $this->customerDao;
     }
 
     /**
      *
-     * @param CustomerDao $customerDao 
+     * @param CustomerDao $customerDao
      */
-    public function setCustomerDao(CustomerDao $customerDao) {
+    public function setCustomerDao(CustomerDao $customerDao)
+    {
         $this->customerDao = $customerDao;
     }
 
     /**
      * Get Customer List
-     * 
+     *
      * Get Customer List with pagination.
-     * 
-     * @param type $noOfRecords
-     * @param type $offset
-     * @param type $sortField
-     * @param type $sortOrder
-     * @param type $activeOnly
-     * @return type 
+     *
+     * @param false $count
+     * @param int|null $offset
+     * @param int|null $limit
+     * @param string $sortField
+     * @param string $sortOrder
+     * @return int|Customer[]
      */
-    public function getCustomerList($limit = 50, $offset = 0, $sortField = 'name', $sortOrder = 'ASC', $activeOnly = true) {
-        return $this->customerDao->getCustomerList($limit, $offset, $sortField, $sortOrder, $activeOnly);
+    public function getCustomerList(
```

Comment:
are we using this. if not remove this method

Reply:
done